DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/DelegateProgressControllerTests.cs

```diff
@@ -138,6 +141,80 @@ string expectedAction
+            DelegateAccessRoute accessedVia,
+            string expectedController,
+            string expectedAction
+        )
+        {
+            // Given
+            const int progressId = 1;
+            const int promptNumber = 1;
+            const string answer = "Test Answer";
+            var formData = new EditDelegateCourseAdminFieldFormData { Answer = answer };
+
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(A<int>._, A<int>._, A<string>._))
+                .DoesNothing();
+
+            // When
+            var result = delegateProgressController.EditDelegateCourseAdminField(
+                formData,
+                promptNumber,
+                progressId,
+                accessedVia,
+                1
+            );
+
+            // Then
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(progressId, promptNumber, answer))
+                .MustHaveHappenedOnceExactly();
+            result.Should().BeRedirectToActionResult().WithControllerName(expectedController)
+                .WithActionName(expectedAction);
+        }
+
+        [Test]
+        public void EditDelegateCourseAdminField_does_not_call_service_with_invalid_model()
+        {
+            // Given
+            const int progressId = 1;
+            const int promptNumber = 1;
+            const string answer = "Test Answer";
+
+            var delegateCourseInfo = new DelegateCourseInfo();
+            var courseAdminField = new CourseAdminFieldWithAnswer(1, "Prompt text", "Answer", null);
+            var courseAdminFields = new List<CourseAdminFieldWithAnswer> { courseAdminField };
+            var attemptStats = new AttemptStats(1, 1);
+            var delegateCourseDetails = new DelegateCourseDetails(delegateCourseInfo, courseAdminFields, attemptStats);
+
+            var formData = new EditDelegateCourseAdminFieldFormData { Answer = answer };
+
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(progressId, promptNumber, answer))
+                .DoesNothing();
+            A.CallTo(() => courseService.GetDelegateCourseProgress(progressId))
+                .Returns(delegateCourseDetails);
+            delegateProgressController.ModelState.AddModelError("Answer", "Test error message");
+
+            // When
+            var result = delegateProgressController.EditDelegateCourseAdminField(
+                formData,
+                promptNumber,
+                progressId,
+                DelegateAccessRoute.CourseDelegates,
+                1
+            );
+
+            // Then
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(A<int>._, A<int>._, A<string>._))
+                .MustNotHaveHappened();
+            result.Should().BeViewResult().ModelAs<EditDelegateCourseAdminFieldViewModel>();
+            delegateProgressController.ModelState.IsValid.Should().BeFalse();

```

Comment:
Don't think we need to check the ModelState is invalid, as it is dealt with by standard MVC controller logic when setting the ModelError

Reply:
Cool, removed

Reply:
Weird, it's gone for good now